sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -647,7 +647,7 @@ def set(self, ts):
         windows=(self._window, ),
         clear_bit=False,
         fire_timestamp=ts,
-        hold_timestamp=self._input_timestamp,
+        hold_timestamp=ts,
```

Comment:
Can we only use fire_timestamp as hold_timestamp when in event time domain?

Reply:
Do you mean to keep the input timestamp if we have a processing time timer? 

As far as I'm concerned, the hold/output timestamp is elevant for event time timers. Processing timers do not operate in event time and thus do not respect this timestamp, no matter what it is set to. Unless there is a good reason for introducing this conditional behavior, I'd keep it as-is.

Reply:
Also noting that having the fire timestamp as the output timestamp is consistent with the Java reference implementation.

Reply:
Just saw the thread on the mailing list. I didn't know that the behavior for processing timers is different. I'm unsure whether that makes sense but I'll adjust it to be consistent with Java.